File: Java/app/src/main/java/net/consentmanager/kmm/cmpsdkdemoapp/MainActivity.java
package net.consentmanager.kmm.cmpsdkdemoapp;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.activity.ComponentActivity;
import androidx.annotation.NonNull;

import net.consentmanager.sdk.CmpManager;
import net.consentmanager.sdk.common.CmpError;
import net.consentmanager.sdk.common.callbacks.CmpLayerAppEventListenerInterface;
import net.consentmanager.sdk.consentlayer.model.CmpConfig;
import net.consentmanager.sdk.consentlayer.model.CmpUIConfig;
import net.consentmanager.sdk.consentlayer.model.CmpUIStrategy;

public class MainActivity extends ComponentActivity implements CmpLayerAppEventListenerInterface {
    private CmpManager cmpManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Configure CMP
        CmpConfig.INSTANCE.setId("Your Code-ID goes here"); // TODO: Replace with your Code-ID
        CmpConfig.INSTANCE.setDomain("delivery.consentmanager.net");
        CmpConfig.INSTANCE.setLanguage("EN");
        CmpConfig.INSTANCE.setAppName("CMDemoAppSDKv2Java");

        // Set UI Strategy
        CmpUIConfig.INSTANCE.setUiStrategy(CmpUIStrategy.POPUP);

        setContentView(new View(this));

        // Initialize CMP Manager
        cmpManager = CMPManagerSingleton.initialize(this, CmpConfig.INSTANCE, this);

        setContentView(R.layout.activity_cmp_demo);
        showCMPDemoScreen();
    }

    private void showCMPDemoScreen() {
        Intent intent = new Intent(this, CMPDemoActivity.class);
        startActivity(intent);
        //finish(); // Uncomment if you want to close MainActivity
    }

    @Override
    public void onOpenRequest() {
        Log.d("CMPLayer", "onOpenRequest");
    }

    @Override
    public void onCloseRequest() {
        Log.d("CMPLayer", "onCloseRequest");
    }

    @Override
    public void onError(@NonNull CmpError error, @NonNull String message) {
        // Handle error event
        Log.e("CMPLayer", "Error: " + error + ", Message: " + message);
    }
}

File: Kotlin/app/src/main/java/net/consentmanager/kmm/cmpsdkdemoapp/MainActivity.kt
package net.consentmanager.kmm.cmpsdkdemoapp

import android.os.Bundle
import android.util.Log
import androidx.activity.OnBackPressedCallback
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.ui.Modifier
import net.consentmanager.sdk.CmpManager
import net.consentmanager.sdk.consentlayer.model.CmpConfig
import net.consentmanager.sdk.consentlayer.model.CmpUIConfig
import net.consentmanager.sdk.consentlayer.model.CmpUIStrategy

class MainActivity : AppCompatActivity() {
    private lateinit var cmpManager: CmpManager
    private var isCmpShowing = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val config = CmpConfig.apply {
            id = "YOUR-CODE-ID" // TODO replace with your own Code-ID from your CMP
            domain = "delivery.consentmanager.net"
            language = "IT"
            appName = "CMDemoAppKotlin"
        }
        CmpUIConfig.uiStrategy = CmpUIStrategy.POPUP

        cmpManager = CmpManager.createInstance(this, config).apply {
            initialize(this@MainActivity)
            withOpenListener { isCmpShowing = true }
            withCloseListener { isCmpShowing = false }
            withErrorCallback { error, message ->
                Log.e("CMPDemoApp", "Error: $error, Message: $message")
            }
        }
        setupBackPressHandler()
        showCMPDemoScreen()
    }

    private fun setupBackPressHandler() {
        onBackPressedDispatcher.addCallback(this, object : OnBackPressedCallback(true) {
            override fun handleOnBackPressed() {
                if (isCmpShowing) {
                    if (CmpUIConfig.backPressCallback?.onBackPressed() != true) {
                        return
                    }
                }
                isEnabled = false // Disable this callback to allow the system back press to proceed
                onBackPressedDispatcher.onBackPressed()
            }
        })
    }

    private fun showCMPDemoScreen() {
        setContent {
            MaterialTheme {
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colorScheme.background
                ) {
                    CMPDemoScreen(cmpManager)
                }
            }
        }
    }
}
